packages/nocodb/src/lib/db/sql-data-mapper/lib/sql/formulav2/formulaQueryBuilderv2.ts

```diff
@@ -659,6 +663,28 @@ export default async function formulaQueryBuilderv2(
       const right = fn(pt.right, null, pt.operator).toQuery();
       let sql = `${left} ${pt.operator} ${right}${colAlias}`;
 
+      // comparing a date with empty string would throw
+      // `ERROR: zero-length delimited identifier` in Postgres
+      if (
+        knex.clientType() === 'pg' &&
+        columnIdToUidt[pt.left.name] === UITypes.Date
+      ) {
+        // The correct way to compare with Date should be using
+        // `IS_AFTER`, `IS_BEFORE`, or `IS_SAME`
+        // This is to prevent empty data returned to UI due to incorrect SQL
+        if (pt.right.value === '') {
+          if (pt.operator === '=') {
+            sql = `${left} IS NULL ${colAlias}`;
+          } else {
+            sql = `${left} IS NOT NULL ${colAlias}`;
+          }
+        } else if (!validateDateWithUnknownFormat(pt.right.value)) {
+          // left tree value is date but right tree value is not date
+          // return true if left tree value is not null, else false
+          sql = `${left} IS NOT NULL ${colAlias}`;
+        }
+      }
```

Comment:
The formula can be `"" = {DatePaid}` as well

Reply:
Good Catch!